bundles/org.openhab.binding.ecotouch/src/main/java/org/openhab/binding/ecotouch/internal/EcoTouchHandler.java

```diff
@@ -0,0 +1,236 @@
+public class EcoTouchHandler extends BaseThingHandler {
+
+    private @Nullable EcoTouchConnector connector = null;
+
+    private final Logger logger = LoggerFactory.getLogger(EcoTouchHandler.class);
+
+    private @Nullable EcoTouchConfiguration config = null;
+
+    private @Nullable ScheduledFuture<?> refreshJob = null;
+
+    public EcoTouchHandler(Thing thing) {
+        super(thing);
+    }
+
+    private void updateChannel(String tag, String value_str) {
+        try {
+            List<EcoTouchTags> ecoTouchTags = EcoTouchTags.fromTag(tag);
+            for (EcoTouchTags ecoTouchTag : ecoTouchTags) {
+                String channel = ecoTouchTag.getCommand();
+                BigDecimal value = ecoTouchTag.decodeValue(value_str);
+                if (ecoTouchTag == EcoTouchTags.TYPE_ADAPT_HEATING) {
+                    // this type needs special treatment
+                    // the following reads: value = value / 2 - 2
+                    value = value.divide(new BigDecimal(2), 1, RoundingMode.UNNECESSARY).subtract(new BigDecimal(2));
+                    QuantityType<?> quantity = new QuantityType<javax.measure.quantity.Temperature>(value, CELSIUS);
+                    updateState(channel, quantity);
+                } else {
+                    if (ecoTouchTag.getUnit() != ONE) {
+                        // this is a quantity type
+                        QuantityType<?> quantity = new QuantityType<>(value, ecoTouchTag.getUnit());
+                        updateState(channel, quantity);
+                    } else {
+                        DecimalType number = new DecimalType(value);
+                        updateState(channel, number);
+                    }
+                }
+            }
+        } catch (Exception e) {
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        var localConnector = connector;
+        if (localConnector == null) {
+            // the binding was not initialized, yet
+            return;
+        }
+        if (command instanceof RefreshType) {
+            // request a refresh of a channel
+            try {
+                EcoTouchTags tag = EcoTouchTags.fromString(channelUID.getId());
+                if (tag != null) {
+                    String valueStr = localConnector.getValue(tag.getTagName());
+                    updateChannel(tag.getTagName(), valueStr);
+                    updateStatus(ThingStatus.ONLINE);
+                }
+            } catch (Exception e) {
+            }
+        } else {
+            // send command to heat pump
+            try {
+                EcoTouchTags ecoTouchTag = EcoTouchTags.fromString(channelUID.getId());
+                if (ecoTouchTag == EcoTouchTags.TYPE_ADAPT_HEATING) {
+                    // this type needs special treatment
+                    QuantityType<?> value = (QuantityType<?>) command;
+                    int raw = Math.round(value.floatValue() * 2 + 4);
```

Comment:
Keep in mind that you need to convert the value with `QuantityType.toValue()` to the unit you expect. `value.floatValue()` will return Fahrenheit if the user uses Fahrenheit.